File: Ahorn/entities/ExitDashBlock.jl
module VivHelperExitDashBlock
using ..Ahorn, Maple

@mapdef Entity "VivHelper/ExitDashBlock" ExitDashBlock(x::Integer, y::Integer, width::Integer=8, height::Integer=8, tiletype::String="3", blendin::Bool=true, canDash::Bool=true, permanent::Bool=true)

const placements = Ahorn.PlacementDict(
    "Exit Dash Block (Viv's Helper)" => Ahorn.EntityPlacement(
        ExitDashBlock,
        "rectangle"
    )
)

Ahorn.editingOptions(entity::ExitDashBlock) = Dict{String, Any}(
    "tiletype" => Ahorn.tiletypeEditingOptions()
)

Ahorn.minimumSize(entity::ExitDashBlock) = 8, 8
Ahorn.resizable(entity::ExitDashBlock) = true, true

Ahorn.selection(entity::ExitDashBlock) = Ahorn.getEntityRectangle(entity)

Ahorn.renderAbs(ctx::Ahorn.Cairo.CairoContext, entity::ExitDashBlock, room::Maple.Room) = Ahorn.drawTileEntity(ctx, room, entity, alpha=0.7)

end

File: Loenn/entities/ExitDashBlock.lua
local fakeTilesHelper = require('helpers.fake_tiles')

local exitDashBlock = {}

exitDashBlock.name = "VivHelper/ExitDashBlock"
exitDashBlock.depth = -13000
exitDashBlock.placements = {
    name = "main",
    data = {
        tiletype = "3",
        blendin = false,
        width = 8,
        height = 8,
        canDash = true,
        permanent = false,
    }
}


exitDashBlock.sprite = fakeTilesHelper.getEntitySpriteFunction("tiletype", "blendin")
exitDashBlock.fieldInformation = fakeTilesHelper.getFieldInformation("tiletype")

return exitDashBlock

